<!DOCTYPE html>
<html lang="zh-CN">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 4.2.1">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">


<link rel="stylesheet" href="/lib/font-awesome/css/all.min.css">
  <link rel="stylesheet" href="//cdn.jsdelivr.net/gh/fancyapps/fancybox@3/dist/jquery.fancybox.min.css">
<script src="//cdn.staticfile.org/jquery/3.4.1/jquery.min.js"></script>
<script src="https://cdn.staticfile.org/layer/3.1.1/layer.min.js"></script>
<script src="https://boycott-ie.github.io/boycott-ie/boycott-ie.js"></script>

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"www.microbiomecat.xyz","root":"/","scheme":"Gemini","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":true,"show_result":true,"style":"default"},"back2top":{"enable":true,"sidebar":true,"scrollpercent":true},"bookmark":{"enable":true,"color":"#222","save":"auto"},"fancybox":true,"mediumzoom":true,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":"valine","storage":true,"lazyload":false,"nav":null,"activeClass":"valine"},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":true,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}},"path":"search.xml"};
  </script>

  <meta name="description" content="摘要 使用 QIIME2 2020-8 进行扩增子测序的实际下机数据 (PE250) 的分析。">
<meta property="og:type" content="article">
<meta property="og:title" content="基于 QIIME2 的扩增子分析教程-实战篇">
<meta property="og:url" content="https://www.microbiomecat.xyz/20201006/QIIME2-based-amplicon-analysis-tutorial(Practical)/index.html">
<meta property="og:site_name" content="MicrobiomeCAT">
<meta property="og:description" content="摘要 使用 QIIME2 2020-8 进行扩增子测序的实际下机数据 (PE250) 的分析。">
<meta property="og:locale" content="zh_CN">
<meta property="og:image" content="https://gitee.com/microbiomecat/image_hosting/raw/master/img/image-20200923152644270.png">
<meta property="og:image" content="https://gitee.com/microbiomecat/image_hosting/raw/master/img/image-20200923153925004.png">
<meta property="og:image" content="https://gitee.com/microbiomecat/image_hosting/raw/master/img/image-20200923155639628.png">
<meta property="og:image" content="https://gitee.com/microbiomecat/image_hosting/raw/master/img/image-20200923172129340.png">
<meta property="og:image" content="https://gitee.com/microbiomecat/image_hosting/raw/master/img/image-20200923172206120.png">
<meta property="og:image" content="https://gitee.com/microbiomecat/image_hosting/raw/master/img/image-20200923172509827.png">
<meta property="og:image" content="https://gitee.com/microbiomecat/image_hosting/raw/master/img/image-20200923173136041.png">
<meta property="og:image" content="https://gitee.com/microbiomecat/image_hosting/raw/master/img/image-20200923175247448.png">
<meta property="og:image" content="https://gitee.com/microbiomecat/image_hosting/raw/master/img/image-20200923175456345.png">
<meta property="og:image" content="c:%5CUsers%5C86199%5CDesktop%5Cqiime2%5Cdata%5Cimage-20200923220047881.png">
<meta property="og:image" content="https://gitee.com/microbiomecat/image_hosting/raw/master/img/image-20200923223303485.png">
<meta property="og:image" content="https://gitee.com/microbiomecat/image_hosting/raw/master/img/image-20200923223411020.png">
<meta property="og:image" content="https://gitee.com/microbiomecat/image_hosting/raw/master/img/image-20200923224132572.png">
<meta property="og:image" content="https://gitee.com/microbiomecat/image_hosting/raw/master/img/image-20200924001539896.png">
<meta property="og:image" content="https://gitee.com/microbiomecat/image_hosting/raw/master/img/image-20200923210736980.png">
<meta property="article:published_time" content="2020-10-06T13:18:56.000Z">
<meta property="article:modified_time" content="2020-10-07T13:08:56.000Z">
<meta property="article:author" content="Sansan Li">
<meta property="article:tag" content="QIIME2">
<meta property="article:tag" content="扩增子">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://gitee.com/microbiomecat/image_hosting/raw/master/img/image-20200923152644270.png">

<link rel="canonical" href="https://www.microbiomecat.xyz/20201006/QIIME2-based-amplicon-analysis-tutorial(Practical)/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'zh-CN'
  };
</script>

  <title>基于 QIIME2 的扩增子分析教程-实战篇 | MicrobiomeCAT</title>
  






  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="切换导航栏">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">MicrobiomeCAT</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
      <p class="site-subtitle" itemprop="description">Microbiome Core Analytical Techniques</p>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
        <i class="fa fa-search fa-fw fa-lg"></i>
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/" rel="section"><i class="fa fa-home fa-fw"></i>首页</a>

  </li>
        <li class="menu-item menu-item-about">

    <a href="/about" rel="section"><i class="fa fa-user fa-fw"></i>关于</a>

  </li>
        <li class="menu-item menu-item-tags">

    <a href="/tags" rel="section"><i class="fa fa-tags fa-fw"></i>标签</a>

  </li>
        <li class="menu-item menu-item-categories">

    <a href="/categories" rel="section"><i class="fa fa-th fa-fw"></i>分类</a>

  </li>
        <li class="menu-item menu-item-archives">

    <a href="/archives" rel="section"><i class="fa fa-archive fa-fw"></i>归档</a>

  </li>
      <li class="menu-item menu-item-search">
        <a role="button" class="popup-trigger"><i class="fa fa-search fa-fw"></i>搜索
        </a>
      </li>
  </ul>
</nav>



  <div class="search-pop-overlay">
    <div class="popup search-popup">
        <div class="search-header">
  <span class="search-icon">
    <i class="fa fa-search"></i>
  </span>
  <div class="search-input-container">
    <input autocomplete="off" autocapitalize="off"
           placeholder="搜索..." spellcheck="false"
           type="search" class="search-input">
  </div>
  <span class="popup-btn-close">
    <i class="fa fa-times-circle"></i>
  </span>
</div>
<div id="search-result">
  <div id="no-result">
    <i class="fa fa-spinner fa-pulse fa-5x fa-fw"></i>
  </div>
</div>

    </div>
  </div>

</div>
    </header>

    
  <a role="button" class="book-mark-link book-mark-link-fixed"></a>


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content post posts-expand">
            

    
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="zh-CN">
    <link itemprop="mainEntityOfPage" href="https://www.microbiomecat.xyz/20201006/QIIME2-based-amplicon-analysis-tutorial(Practical)/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/image/876364972032900262.jpg">
      <meta itemprop="name" content="Sansan Li">
      <meta itemprop="description" content="MicrobiomeCAT -- 微生物组学猫">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="MicrobiomeCAT">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          基于 QIIME2 的扩增子分析教程-实战篇
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">发表于</span>

              <time title="创建时间：2020-10-06 21:18:56" itemprop="dateCreated datePublished" datetime="2020-10-06T21:18:56+08:00">2020-10-06</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">分类于</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/%E6%89%A9%E5%A2%9E%E5%AD%90/" itemprop="url" rel="index"><span itemprop="name">扩增子</span></a>
                </span>
            </span>

          
            <span class="post-meta-item" title="阅读次数" id="busuanzi_container_page_pv" style="display: none;">
              <span class="post-meta-item-icon">
                <i class="fa fa-eye"></i>
              </span>
              <span class="post-meta-item-text">阅读次数：</span>
              <span id="busuanzi_value_page_pv"></span>
            </span>
  
  <span class="post-meta-item">
    
      <span class="post-meta-item-icon">
        <i class="far fa-comment"></i>
      </span>
      <span class="post-meta-item-text">Valine：</span>
    
    <a title="valine" href="/20201006/QIIME2-based-amplicon-analysis-tutorial(Practical)/#valine-comments" itemprop="discussionUrl">
      <span class="post-comments-count valine-comment-count" data-xid="/20201006/QIIME2-based-amplicon-analysis-tutorial(Practical)/" itemprop="commentCount"></span>
    </a>
  </span>
  
  <br>
            <span class="post-meta-item" title="本文字数">
              <span class="post-meta-item-icon">
                <i class="far fa-file-word"></i>
              </span>
                <span class="post-meta-item-text">本文字数：</span>
              <span>20k</span>
            </span>
            <span class="post-meta-item" title="阅读时长">
              <span class="post-meta-item-icon">
                <i class="far fa-clock"></i>
              </span>
                <span class="post-meta-item-text">阅读时长 &asymp;</span>
              <span>18 分钟</span>
            </span>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <h2><span id="摘要">摘要</span></h2>
<p>使用 QIIME2 2020-8 进行扩增子测序的实际下机数据 (PE250) 的分析。</p>
<a id="more"></a>
<hr>
<!-- toc -->
<ul>
<li><a href="#%E4%B8%80-%E8%8E%B7%E5%8F%96%E6%95%B0%E6%8D%AE">一、获取数据</a></li>
<li><a href="#%E4%BA%8C-%E5%8F%82%E6%95%B0%E6%B1%87%E6%80%BB">二、参数汇总</a></li>
<li><a href="#%E4%B8%89-%E5%88%9B%E5%BB%BAmanifest%E6%96%87%E4%BB%B6">三、创建manifest文件</a></li>
<li><a href="#%E5%9B%9B-%E4%B8%80%E4%B8%AA%E6%96%87%E5%BA%93%E7%9A%84%E6%93%8D%E4%BD%9C%E6%BC%94%E7%A4%BA">(四)、一个文库的操作演示</a>
<ul>
<li><a href="#%E6%95%B0%E6%8D%AE%E5%AF%BC%E5%85%A5-qiime2">数据导入 QIIME2</a></li>
<li><a href="#%E5%88%87%E9%99%A4%E5%BC%95%E7%89%A9">切除引物</a></li>
<li><a href="#dada2">DADA2</a></li>
</ul>
</li>
<li><a href="#%E5%9B%9B-%E4%B8%80%E4%B8%AA%E6%88%96%E5%A4%9A%E4%B8%AA%E6%96%87%E5%BA%93">四、一个或多个文库</a></li>
<li><a href="#%E4%BA%94-%E5%90%88%E5%B9%B6%E7%89%B9%E5%BE%81%E5%BA%8F%E5%88%97%E4%B8%8E%E7%89%B9%E5%BE%81%E8%A1%A8%E6%A0%BC">五、合并特征序列与特征表格</a></li>
<li><a href="#%E5%85%AD-%E5%BB%BA%E6%A0%91">六、建树</a></li>
<li><a href="#%E4%B8%83-%E7%89%A9%E7%A7%8D%E6%B3%A8%E9%87%8A">七、物种注释</a></li>
<li><a href="#%E5%85%AB-%E6%A0%B8%E5%BF%83%E5%A4%9A%E6%A0%B7%E6%80%A7%E6%8C%87%E6%95%B0">八、核心多样性指数</a></li>
<li><a href="#%E5%8D%81-qiime2r-%E5%8F%AF%E8%A7%86%E5%8C%96%E6%8E%A2%E7%B4%A2">十、qiime2R 可视化探索</a>
<ul>
<li><a href="#%E6%95%B0%E6%8D%AE%E5%AF%BC%E5%85%A5">数据导入</a></li>
<li><a href="#%E7%BB%98%E5%9B%BE">绘图</a></li>
</ul>
</li>
<li><a href="#%E9%99%84%E6%9E%84%E5%BB%BA%E6%B3%A8%E9%87%8A%E6%95%B0%E6%8D%AE%E5%BA%93">附：构建注释数据库</a></li>
</ul>
<!-- tocstop -->
<h2><span id="一-获取数据">一、获取数据</span></h2>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 重启psn_qiime2容器</span></span><br><span class="line">docker restart psn_qiime2</span><br><span class="line"></span><br><span class="line"><span class="comment"># 交互模式，打开容器的一个终端</span></span><br><span class="line">docker <span class="built_in">exec</span> -it psn_qiime2 bash</span><br><span class="line"></span><br><span class="line"><span class="comment"># 创建目录</span></span><br><span class="line">rm -rf  qiime2-tutorial/qiime2-practical_tutorial/</span><br><span class="line">mkdir -p qiime2-tutorial/qiime2-practical_tutorial/</span><br><span class="line"></span><br><span class="line"><span class="built_in">cd</span> qiime2-tutorial/qiime2-practical_tutorial/</span><br><span class="line"></span><br><span class="line">cp /data/qiime2/data/02_data/practical_rawdata.tar.gz .</span><br><span class="line">cp /data/qiime2/data/02_data/practical_metadata.tsv .</span><br><span class="line"></span><br><span class="line">tar -xzvf practical_rawdata.tar.gz </span><br><span class="line">rm practical_rawdata.tar.gz</span><br><span class="line"></span><br><span class="line">mv rawdata/ 1_rawdata</span><br><span class="line"></span><br><span class="line"><span class="comment"># 假设样本测序数据来自两个文库</span></span><br><span class="line"><span class="built_in">cd</span> 1_rawdata</span><br><span class="line">gzip *</span><br><span class="line">mkdir lib1 lib2</span><br><span class="line">mv B* G* lib1</span><br><span class="line">mv H* S* lib2</span><br><span class="line"><span class="built_in">cd</span> -</span><br><span class="line"></span><br><span class="line"><span class="comment">#  说明：</span></span><br><span class="line"><span class="comment">## 这样便完成了数据层面的准备，如下图。</span></span><br><span class="line"><span class="comment">## 注意，每个序列的原始数据文件都有两个，命名需要符合以下规则：以_分隔样本名和R1或R2，格式为fq.gz (即fastq格式,并进行gzip压缩)。</span></span><br><span class="line"><span class="comment">## 以上的脚本需要根据具体的原始数据的位置、命名进行灵活调整。</span></span><br></pre></td></tr></table></figure>
<img src="https://gitee.com/microbiomecat/image_hosting/raw/master/img/image-20200923152644270.png" alt="image-20200923152644270" style="zoom:200%;">
<h2><span id="二-参数汇总">二、参数汇总</span></h2>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line"></span><br><span class="line"><span class="comment"># 输入引物序列</span></span><br><span class="line">front_f=<span class="string">"GTGCCAGCMGCCGCGGTAA"</span></span><br><span class="line">front_r=<span class="string">"CCGTCAATTCCTTTGAGTTT"</span> </span><br><span class="line"></span><br><span class="line"><span class="comment"># 使用核数</span></span><br><span class="line">thread=2</span><br><span class="line"></span><br><span class="line"><span class="comment"># 扩增子类型与区域</span></span><br><span class="line">region=<span class="string">"16S_V4V5"</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 物种注释数据库</span></span><br><span class="line">db=gg_13_8_99_16S_V4V5_classifier.qza</span><br></pre></td></tr></table></figure>
<h2><span id="三-创建manifest文件">三、创建manifest文件</span></h2>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">echo</span> <span class="string">"sample-id"</span> &gt; sample-id</span><br><span class="line">ls 1_rawdata/*/*R1* | sed <span class="string">"s/.*\///"</span> | cut -f 1 -d <span class="string">"_"</span> &gt;&gt; sample-id</span><br><span class="line"></span><br><span class="line"><span class="built_in">echo</span> <span class="string">"forward-absolute-filepath"</span> &gt; forward-absolute-filepath</span><br><span class="line"><span class="built_in">echo</span> <span class="string">"reverse-absolute-filepath"</span> &gt; reverse-absolute-filepath</span><br><span class="line">ls 1_rawdata/*/*R1* | xargs -i <span class="built_in">echo</span> <span class="string">"<span class="variable">$PWD</span>/&#123;&#125;"</span> &gt;&gt;  forward-absolute-filepath </span><br><span class="line">ls 1_rawdata/*/*R2* | xargs -i <span class="built_in">echo</span> <span class="string">"<span class="variable">$PWD</span>/&#123;&#125;"</span> &gt;&gt;  reverse-absolute-filepath </span><br><span class="line"></span><br><span class="line">paste sample-id forward-absolute-filepath reverse-absolute-filepath &gt; manifest_all</span><br><span class="line">rm sample-id forward-absolute-filepath reverse-absolute-filepath</span><br></pre></td></tr></table></figure>
<p><img src="https://gitee.com/microbiomecat/image_hosting/raw/master/img/image-20200923153925004.png" alt="image-20200923153925004"></p>
<h2><span id="四-一个文库的操作演示">(四)、一个文库的操作演示</span></h2>
<h3><span id="数据导入-qiime2">数据导入 QIIME2</span></h3>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line">i=1_rawdata/lib1</span><br><span class="line">lib=`<span class="built_in">echo</span> <span class="variable">$i</span> | cut -f2 -d<span class="string">"/"</span> `</span><br><span class="line"></span><br><span class="line">manifest_f=`<span class="built_in">echo</span> <span class="variable">$i</span> | sed <span class="string">"s/1_rawdata\//manifest_/"</span>`</span><br><span class="line">head -n1 manifest_all &gt; <span class="variable">$manifest_f</span></span><br><span class="line">ls <span class="variable">$i</span>/*R1* | xargs -i grep &#123;&#125; manifest_all &gt;&gt; <span class="variable">$manifest_f</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 将结果放到2_dada2 里面去</span></span><br><span class="line">mkdir -p 2_dada2</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">qiime tools import \</span><br><span class="line">	--<span class="built_in">type</span> <span class="string">'SampleData[PairedEndSequencesWithQuality]'</span> \</span><br><span class="line">    --input-path <span class="variable">$manifest_f</span> \</span><br><span class="line">    --output-path 2_dada2/demux_<span class="variable">$lib</span>.qza \</span><br><span class="line">    --input-format PairedEndFastqManifestPhred33V2</span><br></pre></td></tr></table></figure>
<p><img src="https://gitee.com/microbiomecat/image_hosting/raw/master/img/image-20200923155639628.png" alt="image-20200923155639628"></p>
<h3><span id="切除引物">切除引物</span></h3>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># cutadapt</span></span><br><span class="line">qiime cutadapt trim-paired  \</span><br><span class="line">	--i-demultiplexed-sequences 2_dada2/demux_<span class="variable">$lib</span>.qza \</span><br><span class="line">    --p-cores <span class="variable">$thread</span> \</span><br><span class="line">    --p-error-rate 0.2 \</span><br><span class="line">    --p-times 1 \</span><br><span class="line">    --p-overlap $[<span class="variable">$&#123;#front_f&#125;</span>*9/10] \</span><br><span class="line">    --p-front<span class="_">-f</span> <span class="variable">$front_f</span> \</span><br><span class="line">    --o-trimmed-sequences 2_dada2/demux_no_primer_<span class="variable">$&#123;lib&#125;</span>_1.qza \</span><br><span class="line">    --p-discard-untrimmed True</span><br><span class="line"> </span><br><span class="line">qiime cutadapt trim-paired  \</span><br><span class="line">	--i-demultiplexed-sequences 2_dada2/demux_no_primer_<span class="variable">$&#123;lib&#125;</span>_1.qza \</span><br><span class="line">    --p-cores <span class="variable">$thread</span> \</span><br><span class="line">    --p-error-rate 0.2 \</span><br><span class="line">    --p-times 1 \</span><br><span class="line">    --p-overlap $[<span class="variable">$&#123;#front_r&#125;</span>*9/10] \</span><br><span class="line">    --p-front-r <span class="variable">$front_r</span> \</span><br><span class="line">    --o-trimmed-sequences 2_dada2/demux_no_primer_<span class="variable">$&#123;lib&#125;</span>.qza \</span><br><span class="line">    --p-discard-untrimmed True</span><br><span class="line"></span><br><span class="line"><span class="comment">## 查看序列长度与质量分布</span></span><br><span class="line">qiime demux summarize \</span><br><span class="line">  --i-data 2_dada2/demux_no_primer_<span class="variable">$lib</span>.qza \</span><br><span class="line">  --o-visualization 2_dada2/demux_no_primer_<span class="variable">$lib</span>.qzv</span><br></pre></td></tr></table></figure>
<p><img src="https://gitee.com/microbiomecat/image_hosting/raw/master/img/image-20200923172129340.png" alt="image-20200923172129340"></p>
<p><img src="https://gitee.com/microbiomecat/image_hosting/raw/master/img/image-20200923172206120.png" alt="image-20200923172206120"></p>
<h3><span id="dada2">DADA2</span></h3>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 计算参数</span></span><br><span class="line"></span><br><span class="line"><span class="comment">## PE300还是250？</span></span><br><span class="line">PE_length=`cut -f2 <span class="variable">$manifest_f</span> | grep <span class="string">"fq.gz"</span> | </span><br><span class="line">head -n 1 | xargs -i zcat &#123;&#125; | </span><br><span class="line">awk <span class="string">'FNR%4==2&#123;print length&#125;'</span> | </span><br><span class="line">sort -n | tail -n 1000 | head -n 1`</span><br><span class="line"></span><br><span class="line">PElength=PE250</span><br><span class="line"><span class="keyword">if</span> [[ <span class="variable">$PE_length</span> -ge 260 ]]; <span class="keyword">then</span></span><br><span class="line">  PElength=PE300</span><br><span class="line"><span class="keyword">fi</span></span><br><span class="line"><span class="keyword">if</span> [[ <span class="variable">$PE_length</span> -lt 260 ]]; <span class="keyword">then</span></span><br><span class="line">  PElength=PE250</span><br><span class="line"><span class="keyword">fi</span></span><br><span class="line"><span class="keyword">if</span> [[ <span class="variable">$PE_length</span> -lt 160 ]]; <span class="keyword">then</span></span><br><span class="line">  PElength=PE150</span><br><span class="line"><span class="keyword">fi</span></span><br><span class="line"></span><br><span class="line"><span class="comment">## 使用记载的dada2参数</span></span><br><span class="line"><span class="keyword">if</span> [ <span class="variable">$region</span> == <span class="string">"16S_V3V4"</span> ]; <span class="keyword">then</span></span><br><span class="line">  <span class="keyword">if</span> [ <span class="variable">$PElength</span> == <span class="string">"PE300"</span> ]; <span class="keyword">then</span></span><br><span class="line">      tlf=230</span><br><span class="line">      tlr=220</span><br><span class="line">  <span class="keyword">fi</span></span><br><span class="line">  <span class="keyword">if</span> [ <span class="variable">$PElength</span> == <span class="string">"PE250"</span> ]; <span class="keyword">then</span></span><br><span class="line">      tlf=220</span><br><span class="line">      tlr=225</span><br><span class="line">  <span class="keyword">fi</span></span><br><span class="line"><span class="keyword">fi</span></span><br><span class="line"><span class="keyword">if</span> [ <span class="variable">$region</span> == <span class="string">"16S_V4"</span> ]; <span class="keyword">then</span></span><br><span class="line">  <span class="keyword">if</span> [ <span class="variable">$PElength</span> == <span class="string">"PE300"</span> ]; <span class="keyword">then</span></span><br><span class="line">      tlf=170</span><br><span class="line">      tlr=130</span><br><span class="line">  <span class="keyword">fi</span></span><br><span class="line">  <span class="keyword">if</span> [ <span class="variable">$PElength</span> == <span class="string">"PE250"</span> ]; <span class="keyword">then</span></span><br><span class="line">      tlf=170</span><br><span class="line">      tlr=130</span><br><span class="line">  <span class="keyword">fi</span></span><br><span class="line"><span class="keyword">fi</span></span><br><span class="line"><span class="keyword">if</span> [ <span class="variable">$region</span> == <span class="string">"16S_V4V5"</span> ]; <span class="keyword">then</span></span><br><span class="line"> <span class="keyword">if</span> [ <span class="variable">$PElength</span> == <span class="string">"PE300"</span> ]; <span class="keyword">then</span></span><br><span class="line">      tlf=230</span><br><span class="line">      tlr=160</span><br><span class="line">  <span class="keyword">fi</span></span><br><span class="line">  <span class="keyword">if</span> [ <span class="variable">$PElength</span> == <span class="string">"PE250"</span> ]; <span class="keyword">then</span></span><br><span class="line">      tlf=220</span><br><span class="line">      tlr=170</span><br><span class="line">  <span class="keyword">fi</span></span><br><span class="line"><span class="keyword">fi</span></span><br><span class="line"><span class="keyword">if</span> [ <span class="variable">$region</span> == <span class="string">"16S_V4V5_archaea"</span> ]; <span class="keyword">then</span></span><br><span class="line">  <span class="keyword">if</span> [ <span class="variable">$PElength</span> == <span class="string">"PE300"</span> ]; <span class="keyword">then</span></span><br><span class="line">      tlf=220</span><br><span class="line">      tlr=180</span><br><span class="line">  <span class="keyword">fi</span></span><br><span class="line">  <span class="keyword">if</span> [ <span class="variable">$PElength</span> == <span class="string">"PE250"</span> ]; <span class="keyword">then</span></span><br><span class="line">      tlf=210</span><br><span class="line">      tlr=190</span><br><span class="line">  <span class="keyword">fi</span></span><br><span class="line"><span class="keyword">fi</span></span><br><span class="line"><span class="keyword">if</span> [ <span class="variable">$region</span> == <span class="string">"ITS_V1"</span> ]; <span class="keyword">then</span></span><br><span class="line">  tlf=0</span><br><span class="line">  tlr=0</span><br><span class="line"><span class="keyword">fi</span></span><br><span class="line"></span><br><span class="line"><span class="built_in">echo</span> <span class="string">"正向序列长度切割为: "</span><span class="variable">$tlf</span></span><br><span class="line"><span class="built_in">echo</span> <span class="string">"反向序列长度切割为: "</span><span class="variable">$tlr</span></span><br><span class="line"></span><br><span class="line"><span class="comment">#  DADA2去噪   </span></span><br><span class="line"><span class="comment">## 引物已去除，故--p-trim-left-f --p-trim-left-r 都设0</span></span><br><span class="line"><span class="comment">## 需要根据扩增区域长度变化范围、序列读长确定--p-trunc-len-f 和--p-trunc-len-r 的值 (我们使用了经验值); </span></span><br><span class="line"><span class="comment">## 由于qiime2里的DADA2要求overlap在12bp及以上，那么切割后的正反序列的长度和要大于目标序列 12bp 以上。</span></span><br><span class="line"><span class="comment">## --p-min-fold-parent-over-abundance 的值，可以根据数据削减情况调整，一般来说，目前扩增子序列的PCR循环数并不算高，故不会产生大量的嵌合体 (如超过30%)，所以如果这一步，你发现去除了超过预期的嵌合体，那么可以尝试提高这个阈值。</span></span><br><span class="line">qiime dada2 denoise-paired \</span><br><span class="line">  --i-demultiplexed-seqs 2_dada2/demux_no_primer_<span class="variable">$lib</span>.qza \</span><br><span class="line">  --p-trim-left<span class="_">-f</span> 0 \</span><br><span class="line">  --p-trim-left-r 0 \</span><br><span class="line">  --p-trunc-len<span class="_">-f</span> <span class="variable">$tlf</span> \</span><br><span class="line">  --p-trunc-len-r <span class="variable">$tlr</span> \</span><br><span class="line">  --p-max-ee<span class="_">-f</span> 2 \</span><br><span class="line">  --p-max-ee-r 4 \</span><br><span class="line">  --p-n-threads <span class="variable">$thread</span> \</span><br><span class="line">  --p-n-reads-learn 1000000 \</span><br><span class="line">  --o-table 2_dada2/table_<span class="variable">$lib</span>.qza \</span><br><span class="line">  --o-representative-sequences 2_dada2/rep-seqs_<span class="variable">$lib</span>.qza \</span><br><span class="line">  --o-denoising-stats 2_dada2/denoising-stats_<span class="variable">$lib</span>.qza </span><br><span class="line"></span><br><span class="line"><span class="comment"># 查看去噪结果统计</span></span><br><span class="line">qiime metadata tabulate \</span><br><span class="line">	--m-input-file 2_dada2/denoising-stats_<span class="variable">$lib</span>.qza \</span><br><span class="line">    --o-visualization 2_dada2/denoising-stats_<span class="variable">$lib</span>.qzv</span><br></pre></td></tr></table></figure>
<p><img src="https://gitee.com/microbiomecat/image_hosting/raw/master/img/image-20200923172509827.png" alt="image-20200923172509827"></p>
<p><img src="https://gitee.com/microbiomecat/image_hosting/raw/master/img/image-20200923173136041.png" alt="image-20200923173136041"></p>
<h2><span id="四-一个或多个文库">四、一个或多个文库</span></h2>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br><span class="line">118</span><br><span class="line">119</span><br><span class="line">120</span><br><span class="line">121</span><br><span class="line">122</span><br><span class="line">123</span><br><span class="line">124</span><br><span class="line">125</span><br><span class="line">126</span><br><span class="line">127</span><br><span class="line">128</span><br><span class="line">129</span><br><span class="line">130</span><br><span class="line">131</span><br><span class="line">132</span><br><span class="line">133</span><br><span class="line">134</span><br><span class="line">135</span><br><span class="line">136</span><br><span class="line">137</span><br><span class="line">138</span><br></pre></td><td class="code"><pre><span class="line"></span><br><span class="line"><span class="keyword">for</span> i <span class="keyword">in</span> $(ls -d 1_rawdata/* )</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">    <span class="comment"># i=1_rawdata/lib1</span></span><br><span class="line">    lib=`<span class="built_in">echo</span> <span class="variable">$i</span> | cut -f2 -d<span class="string">"/"</span> `</span><br><span class="line"></span><br><span class="line">    manifest_f=`<span class="built_in">echo</span> <span class="variable">$i</span> | sed <span class="string">"s/1_rawdata\//manifest_/"</span>`</span><br><span class="line">    head -n1 manifest_all &gt; <span class="variable">$manifest_f</span></span><br><span class="line">    ls <span class="variable">$i</span>/*R1* | xargs -i grep &#123;&#125; manifest_all &gt;&gt; <span class="variable">$manifest_f</span></span><br><span class="line"></span><br><span class="line">    <span class="comment"># 将结果放到2_dada2 里面去</span></span><br><span class="line">    mkdir -p 2_dada2</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">    qiime tools import \</span><br><span class="line">        --<span class="built_in">type</span> <span class="string">'SampleData[PairedEndSequencesWithQuality]'</span> \</span><br><span class="line">        --input-path <span class="variable">$manifest_f</span> \</span><br><span class="line">        --output-path 2_dada2/demux_<span class="variable">$lib</span>.qza \</span><br><span class="line">        --input-format PairedEndFastqManifestPhred33V2</span><br><span class="line"></span><br><span class="line">    <span class="comment"># cutadapt</span></span><br><span class="line">    qiime cutadapt trim-paired  \</span><br><span class="line">        --i-demultiplexed-sequences 2_dada2/demux_<span class="variable">$lib</span>.qza \</span><br><span class="line">        --p-cores <span class="variable">$thread</span> \</span><br><span class="line">        --p-error-rate 0.2 \</span><br><span class="line">        --p-times 1 \</span><br><span class="line">        --p-overlap $[<span class="variable">$&#123;#front_f&#125;</span>*9/10] \</span><br><span class="line">        --p-front<span class="_">-f</span> <span class="variable">$front_f</span> \</span><br><span class="line">        --o-trimmed-sequences 2_dada2/demux_no_primer_<span class="variable">$&#123;lib&#125;</span>_1.qza \</span><br><span class="line">        --p-discard-untrimmed True</span><br><span class="line"></span><br><span class="line">    qiime cutadapt trim-paired  \</span><br><span class="line">        --i-demultiplexed-sequences 2_dada2/demux_no_primer_<span class="variable">$&#123;lib&#125;</span>_1.qza \</span><br><span class="line">        --p-cores <span class="variable">$thread</span> \</span><br><span class="line">        --p-error-rate 0.2 \</span><br><span class="line">        --p-times 1 \</span><br><span class="line">        --p-overlap $[<span class="variable">$&#123;#front_r&#125;</span>*9/10] \</span><br><span class="line">        --p-front-r <span class="variable">$front_r</span> \</span><br><span class="line">        --o-trimmed-sequences 2_dada2/demux_no_primer_<span class="variable">$&#123;lib&#125;</span>.qza \</span><br><span class="line">        --p-discard-untrimmed True</span><br><span class="line"></span><br><span class="line">    <span class="comment">## 查看序列长度与质量分布</span></span><br><span class="line">    qiime demux summarize \</span><br><span class="line">      --i-data 2_dada2/demux_no_primer_<span class="variable">$lib</span>.qza \</span><br><span class="line">      --o-visualization 2_dada2/demux_no_primer_<span class="variable">$lib</span>.qzv</span><br><span class="line"></span><br><span class="line">    <span class="comment"># 计算参数</span></span><br><span class="line"></span><br><span class="line">    <span class="comment">## PE300还是250？</span></span><br><span class="line">    PE_length=`cut -f2 <span class="variable">$manifest_f</span> |</span><br><span class="line">    grep <span class="string">"fq.gz"</span> |head -n 1 | </span><br><span class="line">    xargs -i zcat &#123;&#125; | awk <span class="string">'FNR%4==2&#123;print length&#125;'</span> |</span><br><span class="line">    sort -n | tail -n 1000 | head -n 1`</span><br><span class="line"></span><br><span class="line">    PElength=PE250</span><br><span class="line">    <span class="keyword">if</span> [[ <span class="variable">$PE_length</span> -ge 260 ]]; <span class="keyword">then</span></span><br><span class="line">      PElength=PE300</span><br><span class="line">    <span class="keyword">fi</span></span><br><span class="line">    <span class="keyword">if</span> [[ <span class="variable">$PE_length</span> -lt 260 ]]; <span class="keyword">then</span></span><br><span class="line">      PElength=PE250</span><br><span class="line">    <span class="keyword">fi</span></span><br><span class="line">    <span class="keyword">if</span> [[ <span class="variable">$PE_length</span> -lt 160 ]]; <span class="keyword">then</span></span><br><span class="line">      PElength=PE150</span><br><span class="line">    <span class="keyword">fi</span></span><br><span class="line"></span><br><span class="line">    <span class="comment">## 使用记载的dada2参数</span></span><br><span class="line">    <span class="keyword">if</span> [ <span class="variable">$region</span> == <span class="string">"16S_V3V4"</span> ]; <span class="keyword">then</span></span><br><span class="line">      <span class="keyword">if</span> [ <span class="variable">$PElength</span> == <span class="string">"PE300"</span> ]; <span class="keyword">then</span></span><br><span class="line">          tlf=230</span><br><span class="line">          tlr=220</span><br><span class="line">      <span class="keyword">fi</span></span><br><span class="line">      <span class="keyword">if</span> [ <span class="variable">$PElength</span> == <span class="string">"PE250"</span> ]; <span class="keyword">then</span></span><br><span class="line">          tlf=220</span><br><span class="line">          tlr=225</span><br><span class="line">      <span class="keyword">fi</span></span><br><span class="line">    <span class="keyword">fi</span></span><br><span class="line">    <span class="keyword">if</span> [ <span class="variable">$region</span> == <span class="string">"16S_V4"</span> ]; <span class="keyword">then</span></span><br><span class="line">      <span class="keyword">if</span> [ <span class="variable">$PElength</span> == <span class="string">"PE300"</span> ]; <span class="keyword">then</span></span><br><span class="line">          tlf=170</span><br><span class="line">          tlr=130</span><br><span class="line">      <span class="keyword">fi</span></span><br><span class="line">      <span class="keyword">if</span> [ <span class="variable">$PElength</span> == <span class="string">"PE250"</span> ]; <span class="keyword">then</span></span><br><span class="line">          tlf=170</span><br><span class="line">          tlr=130</span><br><span class="line">      <span class="keyword">fi</span></span><br><span class="line">    <span class="keyword">fi</span></span><br><span class="line">    <span class="keyword">if</span> [ <span class="variable">$region</span> == <span class="string">"16S_V4V5"</span> ]; <span class="keyword">then</span></span><br><span class="line">     <span class="keyword">if</span> [ <span class="variable">$PElength</span> == <span class="string">"PE300"</span> ]; <span class="keyword">then</span></span><br><span class="line">          tlf=230</span><br><span class="line">          tlr=160</span><br><span class="line">      <span class="keyword">fi</span></span><br><span class="line">      <span class="keyword">if</span> [ <span class="variable">$PElength</span> == <span class="string">"PE250"</span> ]; <span class="keyword">then</span></span><br><span class="line">          tlf=220</span><br><span class="line">          tlr=170</span><br><span class="line">      <span class="keyword">fi</span></span><br><span class="line">    <span class="keyword">fi</span></span><br><span class="line">    <span class="keyword">if</span> [ <span class="variable">$region</span> == <span class="string">"16S_V4V5_archaea"</span> ]; <span class="keyword">then</span></span><br><span class="line">      <span class="keyword">if</span> [ <span class="variable">$PElength</span> == <span class="string">"PE300"</span> ]; <span class="keyword">then</span></span><br><span class="line">          tlf=220</span><br><span class="line">          tlr=180</span><br><span class="line">      <span class="keyword">fi</span></span><br><span class="line">      <span class="keyword">if</span> [ <span class="variable">$PElength</span> == <span class="string">"PE250"</span> ]; <span class="keyword">then</span></span><br><span class="line">          tlf=210</span><br><span class="line">          tlr=190</span><br><span class="line">      <span class="keyword">fi</span></span><br><span class="line">    <span class="keyword">fi</span></span><br><span class="line">    <span class="keyword">if</span> [ <span class="variable">$region</span> == <span class="string">"ITS_V1"</span> ]; <span class="keyword">then</span></span><br><span class="line">      tlf=0</span><br><span class="line">      tlr=0</span><br><span class="line">    <span class="keyword">fi</span></span><br><span class="line"></span><br><span class="line">    <span class="built_in">echo</span> <span class="string">"正向序列长度切割为: "</span><span class="variable">$tlf</span></span><br><span class="line">    <span class="built_in">echo</span> <span class="string">"反向序列长度切割为: "</span><span class="variable">$tlr</span></span><br><span class="line"></span><br><span class="line">    <span class="comment">#  DADA2去噪   </span></span><br><span class="line">    <span class="comment">## 引物已去除，故--p-trim-left-f --p-trim-left-r 都设0</span></span><br><span class="line">    <span class="comment">## 需要根据扩增区域长度变化范围、序列读长确定--p-trunc-len-f 和--p-trunc-len-r 的值 (我们使用了经验值); </span></span><br><span class="line">    <span class="comment">## 由于qiime2里的DADA2要求overlap在12bp及以上，那么切割后的正反序列的长度和要大于目标序列 12bp 以上。</span></span><br><span class="line">    <span class="comment">## --p-min-fold-parent-over-abundance 的值，可以根据数据削减情况调整，一般来说，目前扩增子序列的PCR循环数并不算高，故不会产生大量的嵌合体 (如超过30%)，所以如果这一步，你发现去除了超过预期的嵌合体，那么可以尝试提高这个阈值。</span></span><br><span class="line">    qiime dada2 denoise-paired \</span><br><span class="line">      --i-demultiplexed-seqs 2_dada2/demux_no_primer_<span class="variable">$lib</span>.qza \</span><br><span class="line">      --p-trim-left<span class="_">-f</span> 0 \</span><br><span class="line">      --p-trim-left-r 0 \</span><br><span class="line">      --p-trunc-len<span class="_">-f</span> <span class="variable">$tlf</span> \</span><br><span class="line">      --p-trunc-len-r <span class="variable">$tlr</span> \</span><br><span class="line">      --p-max-ee<span class="_">-f</span> 2 \</span><br><span class="line">      --p-max-ee-r 4 \</span><br><span class="line">      --p-n-threads <span class="variable">$thread</span> \</span><br><span class="line">      --p-n-reads-learn 1000000 \</span><br><span class="line">      --o-table 2_dada2/table_<span class="variable">$lib</span>.qza \</span><br><span class="line">      --o-representative-sequences 2_dada2/rep-seqs_<span class="variable">$lib</span>.qza \</span><br><span class="line">      --o-denoising-stats 2_dada2/denoising-stats_<span class="variable">$lib</span>.qza </span><br><span class="line"></span><br><span class="line">    <span class="comment"># 查看去噪结果统计</span></span><br><span class="line">    qiime metadata tabulate \</span><br><span class="line">        --m-input-file 2_dada2/denoising-stats_<span class="variable">$lib</span>.qza \</span><br><span class="line">        --o-visualization 2_dada2/denoising-stats_<span class="variable">$lib</span>.qzv </span><br><span class="line"><span class="keyword">done</span></span><br></pre></td></tr></table></figure>
<img src="https://gitee.com/microbiomecat/image_hosting/raw/master/img/image-20200923175247448.png" alt="image-20200923175247448" style="zoom:150%;">
<img src="https://gitee.com/microbiomecat/image_hosting/raw/master/img/image-20200923175456345.png" alt="image-20200923175456345" style="zoom:150%;">
<h2><span id="五-合并特征序列与特征表格">五、合并特征序列与特征表格</span></h2>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br></pre></td><td class="code"><pre><span class="line">mkdir -p 3_ASV </span><br><span class="line"></span><br><span class="line"><span class="comment"># 合并表格</span></span><br><span class="line">qiime feature-table merge  \</span><br><span class="line">	--i-tables 2_dada2/table_*.qza  \</span><br><span class="line">	--p-overlap-method <span class="string">'sum'</span> \</span><br><span class="line">	--o-merged-table 3_ASV/table.qza &amp;</span><br><span class="line"></span><br><span class="line"><span class="comment"># 合并序列</span></span><br><span class="line">qiime feature-table merge-seqs \</span><br><span class="line">	--i-data 2_dada2/rep-seqs_*.qza \</span><br><span class="line">	--o-merged-data 3_ASV/rep-seqs.qza &amp;</span><br><span class="line"></span><br><span class="line"><span class="built_in">wait</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># </span></span><br><span class="line">qiime feature-table summarize \</span><br><span class="line">	--i-table 3_ASV/table.qza \</span><br><span class="line">	--o-visualization 3_ASV/table.qzv \</span><br><span class="line">	--m-sample-metadata-file practical_metadata.tsv &amp;</span><br><span class="line"></span><br><span class="line">qiime feature-table tabulate-seqs \</span><br><span class="line">	--i-data 3_ASV/rep-seqs.qza \</span><br><span class="line">	--o-visualization 3_ASV/rep-seqs.qzv &amp;</span><br><span class="line"></span><br><span class="line"><span class="comment"># 获得fasta格式的特征序列文件和biom格式的特征表格文件</span></span><br><span class="line"></span><br><span class="line">qiime tools <span class="built_in">export</span> \</span><br><span class="line">	--input-path 3_ASV/table.qza \</span><br><span class="line">    --output-path 3_ASV/table &amp;</span><br><span class="line"></span><br><span class="line">qiime tools <span class="built_in">export</span> \</span><br><span class="line">	--input-path 3_ASV/rep-seqs.qza \</span><br><span class="line">	--output-path 3_ASV/rep-seqs &amp;</span><br><span class="line"></span><br><span class="line"><span class="built_in">wait</span></span><br><span class="line"></span><br><span class="line">cp 3_ASV/rep-seqs/dna-sequences.fasta 3_ASV/rep_seqs.fa</span><br><span class="line"></span><br><span class="line">cp 3_ASV/table/feature-table.biom 3_ASV/table.biom</span><br></pre></td></tr></table></figure>
<h2><span id="六-建树">六、建树</span></h2>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">qiime phylogeny align-to-tree-mafft-fasttree \</span><br><span class="line">	--p-n-threads <span class="variable">$thread</span> \</span><br><span class="line">	--i-sequences 3_ASV/rep-seqs.qza \</span><br><span class="line">	--o-alignment 3_ASV/aligned-rep-seqs.qza \</span><br><span class="line">	--o-masked-alignment 3_ASV/masked-aligned-rep-seqs.qza \</span><br><span class="line">	--o-tree 3_ASV/unrooted-tree.qza \</span><br><span class="line">	--o-rooted-tree 3_ASV/rooted-tree.qza</span><br></pre></td></tr></table></figure>
<p><img src="C:%5CUsers%5C86199%5CDesktop%5Cqiime2%5Cdata%5Cimage-20200923220047881.png" alt="image-20200923220047881"></p>
<h2><span id="七-物种注释">七、物种注释</span></h2>
<p>请先下载数据库 <a href="https://microbiomecat.lanzous.com/ijMpcgvikpa%E3%80%82%E5%B9%B6%E5%B0%86%E5%85%B6%E6%94%BE%E5%88%B0%E5%B7%A5%E4%BD%9C%E6%96%87%E4%BB%B6%E5%A4%B9" target="_blank" rel="noopener">https://microbiomecat.lanzous.com/ijMpcgvikpa。并将其放到工作文件夹</a> (qiime2-tutorial) 中</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 解压 数据库</span></span><br><span class="line">unzip  ../classifier.zip; rm ../classifier.zip</span><br><span class="line">mv classifier.qza gg_13_8_99_16S_V4V5_classifier.qza</span><br><span class="line"></span><br><span class="line"><span class="comment"># 物种注释</span></span><br><span class="line">mkdir -p  4_taxonomy</span><br><span class="line"></span><br><span class="line">qiime feature-classifier classify-sklearn \</span><br><span class="line">	--p-n-jobs <span class="variable">$thread</span> \</span><br><span class="line">	--i-classifier <span class="variable">$db</span> \</span><br><span class="line">    --i-reads 3_ASV/rep-seqs.qza \</span><br><span class="line">    --o-classification 4_taxonomy/taxonomy.qza</span><br><span class="line"></span><br><span class="line">qiime metadata tabulate \</span><br><span class="line">	--m-input-file 4_taxonomy/taxonomy.qza \</span><br><span class="line">	--o-visualization 4_taxonomy/taxonomy.qzv &amp;</span><br><span class="line">  </span><br><span class="line">qiime taxa barplot \</span><br><span class="line">	--i-table 3_ASV/table.qza \</span><br><span class="line">    --i-taxonomy 4_taxonomy/taxonomy.qza \</span><br><span class="line">    --m-metadata-file practical_metadata.tsv \</span><br><span class="line">    --o-visualization 4_taxonomy/taxa-bar-plots.qzv</span><br></pre></td></tr></table></figure>
<p><img src="https://gitee.com/microbiomecat/image_hosting/raw/master/img/image-20200923223303485.png" alt="image-20200923223303485"></p>
<p><img src="https://gitee.com/microbiomecat/image_hosting/raw/master/img/image-20200923223411020.png" alt="image-20200923223411020"></p>
<h2><span id="八-核心多样性指数">八、核心多样性指数</span></h2>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">rm -r -f 5_core-metrics-results</span><br><span class="line">qiime diversity core-metrics-phylogenetic \</span><br><span class="line">	--i-phylogeny 3_ASV/rooted-tree.qza \</span><br><span class="line">    --i-table 3_ASV/table.qza \</span><br><span class="line">    --p-sampling-depth 300 \</span><br><span class="line">    --m-metadata-file practical_metadata.tsv \</span><br><span class="line">     --p-n-jobs-or-threads <span class="variable">$thread</span> \</span><br><span class="line">    --output-dir 5_core-metrics-results</span><br></pre></td></tr></table></figure>
<img src="https://gitee.com/microbiomecat/image_hosting/raw/master/img/image-20200923224132572.png" alt="image-20200923224132572" style="zoom:150%;">
<h2><span id="十-qiime2r-可视化探索">十、qiime2R 可视化探索</span></h2>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 切换conda环境</span></span><br><span class="line">conda deactivate</span><br><span class="line">conda activate R4.0</span><br><span class="line"></span><br><span class="line"><span class="comment"># 调整 metadata 注意下面这是一行，直接复制可能会有问题</span></span><br><span class="line">sed <span class="string">'2i\#q2:types\tcategorical\tcategorical\tcategorical\tnumeric\tcategorical'</span> practical_metadata.tsv &gt;  practical_metadata_new.tsv   </span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment"># 打开R</span></span><br><span class="line">R</span><br></pre></td></tr></table></figure>
<p>以下为R编程环境中脚本</p>
<h3><span id="数据导入">数据导入</span></h3>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">###############################################################</span></span><br><span class="line"><span class="comment"># 加载R包</span></span><br><span class="line"><span class="keyword">library</span>(qiime2R)</span><br><span class="line"> <span class="keyword">library</span>(dplyr)</span><br><span class="line"> </span><br><span class="line"><span class="comment">###############################################################</span></span><br><span class="line"><span class="comment"># 读取QZA</span></span><br><span class="line">SVs&lt;-read_qza(<span class="string">"3_ASV/table.qza"</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment">## 查看帮助</span></span><br><span class="line">?read_qza</span><br><span class="line"></span><br><span class="line"><span class="comment">## SVs 是一个list,查看list的元素都是啥？</span></span><br><span class="line"> names(SVs)</span><br><span class="line"></span><br><span class="line"><span class="comment">## 查看我们导入的这个QZA是什么类型的artifact</span></span><br><span class="line">SVs$type</span><br><span class="line"></span><br><span class="line"><span class="comment">## 我们最关注的，其实还是丰度，看下内容</span></span><br><span class="line">SVs$data[<span class="number">1</span>:<span class="number">5</span>,<span class="number">1</span>:<span class="number">5</span>] </span><br><span class="line"></span><br><span class="line"><span class="comment">###############################################################</span></span><br><span class="line"><span class="comment"># 读取元数据 就是分组表哦</span></span><br><span class="line">metadata&lt;-read_q2metadata(<span class="string">"practical_metadata_new.tsv"</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># 查看前几行</span></span><br><span class="line">head(metadata) </span><br><span class="line"></span><br><span class="line"><span class="comment">################################################################</span></span><br><span class="line"><span class="comment"># 读取taxonomy文件</span></span><br><span class="line">taxonomy&lt;-read_qza(<span class="string">"4_taxonomy/taxonomy.qza"</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># 查看前几行</span></span><br><span class="line">head(taxonomy$data)</span><br><span class="line"></span><br><span class="line"><span class="comment">################################################################</span></span><br><span class="line"><span class="comment"># 直接导入phyloseq</span></span><br><span class="line">physeq&lt;-qza_to_phyloseq(</span><br><span class="line">    features=<span class="string">"3_ASV/table.qza"</span>,</span><br><span class="line">    tree=<span class="string">"3_ASV/rooted-tree.qza"</span>,</span><br><span class="line">    taxonomy=<span class="string">"4_taxonomy/taxonomy.qza"</span>,</span><br><span class="line">    metadata = <span class="string">"practical_metadata_new.tsv"</span></span><br><span class="line">    )</span><br></pre></td></tr></table></figure>
<img src="https://gitee.com/microbiomecat/image_hosting/raw/master/img/image-20200924001539896.png" alt="image-20200924001539896" style="zoom: 200%;">
<h3><span id="绘图">绘图</span></h3>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br><span class="line">118</span><br><span class="line">119</span><br><span class="line">120</span><br><span class="line">121</span><br><span class="line">122</span><br><span class="line">123</span><br><span class="line">124</span><br><span class="line">125</span><br><span class="line">126</span><br><span class="line">127</span><br><span class="line">128</span><br><span class="line">129</span><br><span class="line">130</span><br><span class="line">131</span><br><span class="line">132</span><br><span class="line">133</span><br><span class="line">134</span><br><span class="line">135</span><br><span class="line">136</span><br><span class="line">137</span><br><span class="line">138</span><br><span class="line">139</span><br><span class="line">140</span><br><span class="line">141</span><br><span class="line">142</span><br><span class="line">143</span><br><span class="line">144</span><br><span class="line">145</span><br><span class="line">146</span><br><span class="line">147</span><br><span class="line">148</span><br><span class="line">149</span><br><span class="line">150</span><br><span class="line">151</span><br><span class="line">152</span><br><span class="line">153</span><br><span class="line">154</span><br><span class="line">155</span><br><span class="line">156</span><br><span class="line">157</span><br><span class="line">158</span><br><span class="line">159</span><br><span class="line">160</span><br><span class="line">161</span><br><span class="line">162</span><br><span class="line">163</span><br><span class="line">164</span><br><span class="line">165</span><br><span class="line">166</span><br><span class="line">167</span><br><span class="line">168</span><br><span class="line">169</span><br><span class="line">170</span><br><span class="line">171</span><br><span class="line">172</span><br><span class="line">173</span><br><span class="line">174</span><br><span class="line">175</span><br><span class="line">176</span><br><span class="line">177</span><br><span class="line">178</span><br><span class="line">179</span><br><span class="line">180</span><br><span class="line">181</span><br><span class="line">182</span><br><span class="line">183</span><br><span class="line">184</span><br><span class="line">185</span><br><span class="line">186</span><br><span class="line">187</span><br><span class="line">188</span><br><span class="line">189</span><br><span class="line">190</span><br><span class="line">191</span><br><span class="line">192</span><br><span class="line">193</span><br><span class="line">194</span><br><span class="line">195</span><br><span class="line">196</span><br><span class="line">197</span><br><span class="line">198</span><br><span class="line">199</span><br><span class="line">200</span><br><span class="line">201</span><br><span class="line">202</span><br><span class="line">203</span><br><span class="line">204</span><br><span class="line">205</span><br><span class="line">206</span><br><span class="line">207</span><br><span class="line">208</span><br><span class="line">209</span><br><span class="line">210</span><br><span class="line">211</span><br><span class="line">212</span><br><span class="line">213</span><br><span class="line">214</span><br><span class="line">215</span><br><span class="line">216</span><br><span class="line">217</span><br><span class="line">218</span><br><span class="line">219</span><br><span class="line">220</span><br><span class="line">221</span><br><span class="line">222</span><br><span class="line">223</span><br><span class="line">224</span><br><span class="line">225</span><br><span class="line">226</span><br><span class="line">227</span><br><span class="line">228</span><br><span class="line">229</span><br><span class="line">230</span><br><span class="line">231</span><br><span class="line">232</span><br><span class="line">233</span><br><span class="line">234</span><br><span class="line">235</span><br><span class="line">236</span><br><span class="line">237</span><br><span class="line">238</span><br><span class="line">239</span><br><span class="line">240</span><br><span class="line">241</span><br><span class="line">242</span><br><span class="line">243</span><br><span class="line">244</span><br><span class="line">245</span><br><span class="line">246</span><br><span class="line">247</span><br><span class="line">248</span><br><span class="line">249</span><br><span class="line">250</span><br><span class="line">251</span><br><span class="line">252</span><br><span class="line">253</span><br><span class="line">254</span><br><span class="line">255</span><br><span class="line">256</span><br><span class="line">257</span><br><span class="line">258</span><br><span class="line">259</span><br><span class="line">260</span><br><span class="line">261</span><br><span class="line">262</span><br><span class="line">263</span><br><span class="line">264</span><br><span class="line">265</span><br><span class="line">266</span><br><span class="line">267</span><br><span class="line">268</span><br><span class="line">269</span><br><span class="line">270</span><br><span class="line">271</span><br><span class="line">272</span><br><span class="line">273</span><br><span class="line">274</span><br><span class="line">275</span><br><span class="line">276</span><br><span class="line">277</span><br><span class="line">278</span><br><span class="line">279</span><br><span class="line">280</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 安装tidyverse包</span></span><br><span class="line">install.packages(<span class="string">"tidyverse"</span>)</span><br><span class="line">install.packages(<span class="string">"gplots"</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># 载入R包</span></span><br><span class="line"><span class="keyword">library</span>(tidyverse)</span><br><span class="line"><span class="keyword">library</span>(qiime2R)</span><br><span class="line"></span><br><span class="line"><span class="comment"># 读取数据</span></span><br><span class="line">metadata&lt;-read_q2metadata(<span class="string">"practical_metadata_new.tsv"</span>)</span><br><span class="line">shannon&lt;-read_qza(<span class="string">"5_core-metrics-results/shannon_vector.qza"</span>) </span><br><span class="line"></span><br><span class="line">names(shannon)</span><br><span class="line"></span><br><span class="line">shannon&lt;-shannon$data %&gt;% rownames_to_column(<span class="string">"SampleID"</span>) </span><br><span class="line"></span><br><span class="line">head(shannon)</span><br><span class="line"></span><br><span class="line"><span class="comment">## VENN图</span></span><br><span class="line">pdf(<span class="string">'venn.pdf'</span>,onefile=<span class="literal">F</span>,width=<span class="number">10</span>,bg=<span class="string">"white"</span>)</span><br><span class="line">	gplots::venn(list(metadata=metadata$SampleID,</span><br><span class="line">                      shannon=shannon$SampleID))</span><br><span class="line">dev.off()</span><br><span class="line"></span><br><span class="line"><span class="comment">## 折线图</span></span><br><span class="line">metadata&lt;- metadata %&gt;% </span><br><span class="line">  left_join(shannon)</span><br><span class="line"></span><br><span class="line">head(metadata)</span><br><span class="line"></span><br><span class="line">metadata %&gt;%</span><br><span class="line">  filter(!is.na(metadata$shannon_entropy)) %&gt;%</span><br><span class="line">  ggplot(aes(x=`Treat2`, y=shannon_entropy, color=`Treat1`)) +</span><br><span class="line">  stat_summary(geom=<span class="string">"errorbar"</span>, fun.data=mean_se, width=<span class="number">0</span>) +</span><br><span class="line">  stat_summary(geom=<span class="string">"line"</span>, fun.data=mean_se) +</span><br><span class="line">  stat_summary(geom=<span class="string">"point"</span>, fun.data=mean_se) +</span><br><span class="line">  xlab(<span class="string">"Days"</span>) +</span><br><span class="line">  ylab(<span class="string">"Shannon Diversity"</span>) +</span><br><span class="line">  theme_q2r() + </span><br><span class="line">  scale_color_viridis_d(name=<span class="string">"Sampling Region"</span>) </span><br><span class="line">  ggsave(<span class="string">"Shannon_by_time.pdf"</span>, </span><br><span class="line">         height=<span class="number">3</span>, width=<span class="number">4</span>, device=<span class="string">"pdf"</span>) </span><br><span class="line"></span><br><span class="line"><span class="comment">## 柱状图</span></span><br><span class="line">metadata %&gt;%</span><br><span class="line">  filter(!is.na(metadata$shannon_entropy)) %&gt;%</span><br><span class="line">  ggplot(aes(x=`Treat1`, y=shannon_entropy, fill=`Treat1`)) +</span><br><span class="line">  stat_summary(geom=<span class="string">"bar"</span>, fun.data=mean_se, color=<span class="string">"black"</span>) + </span><br><span class="line">  geom_jitter(shape=<span class="number">21</span>, width=<span class="number">0.2</span>, height=<span class="number">0</span>) +</span><br><span class="line">  coord_cartesian(ylim=c(<span class="number">2</span>,<span class="number">7</span>)) + <span class="comment"># adjust y-axis</span></span><br><span class="line"> <span class="comment"># facet_grid(~`Treat1`) +</span></span><br><span class="line">  xlab(<span class="string">"regions"</span>) +</span><br><span class="line">  ylab(<span class="string">"Shannon Diversity"</span>) +</span><br><span class="line">  theme_q2r() +</span><br><span class="line">  scale_fill_manual(values=c(<span class="string">"cornflowerblue"</span>,</span><br><span class="line">                             <span class="string">"indianred"</span>,<span class="string">"green"</span>,<span class="string">"yellow"</span>)) +</span><br><span class="line">  theme(legend.position=<span class="string">"none"</span>) </span><br><span class="line">  ggsave(<span class="string">"Shannon_by_abx.pdf"</span>, height=<span class="number">3</span>,</span><br><span class="line">         width=<span class="number">4</span>, device=<span class="string">"pdf"</span>) </span><br><span class="line"></span><br><span class="line"><span class="comment">## 散点图</span></span><br><span class="line">uwunifrac&lt;-read_qza(</span><br><span class="line">    <span class="string">"5_core-metrics-results/unweighted_unifrac_pcoa_results.qza"</span>)</span><br><span class="line"></span><br><span class="line">uwunifrac$data$Vectors %&gt;%</span><br><span class="line">  select(SampleID, PC1, PC2) %&gt;%</span><br><span class="line">  left_join(metadata) %&gt;%</span><br><span class="line">  left_join(shannon) %&gt;%</span><br><span class="line">  ggplot(aes(x=PC1, y=PC2, color=`Treat1`,</span><br><span class="line">             shape=`Treat1`, size=shannon_entropy)) +</span><br><span class="line">  geom_point(alpha=<span class="number">0.5</span>) + </span><br><span class="line">  theme_q2r() +</span><br><span class="line">  scale_shape_manual(values=c(<span class="number">16</span>,<span class="number">1</span>,<span class="number">21</span>,<span class="number">22</span>), name=<span class="string">"regions"</span>) +</span><br><span class="line">  scale_size_continuous(name=<span class="string">"Shannon Diversity"</span>) +</span><br><span class="line">  scale_color_discrete(name=<span class="string">"regions"</span>)</span><br><span class="line">  ggsave(<span class="string">"PCoA.pdf"</span>, height=<span class="number">4</span>, width=<span class="number">5</span>, device=<span class="string">"pdf"</span>) </span><br><span class="line"></span><br><span class="line"><span class="comment">## 热图</span></span><br><span class="line"></span><br><span class="line"></span><br><span class="line">make_percent&lt;-<span class="keyword">function</span>(features)&#123;</span><br><span class="line">  features&lt;-apply(features,<span class="number">2</span>, <span class="keyword">function</span>(x)&#123;<span class="number">100</span>*(x/sum(x))&#125;)</span><br><span class="line">  <span class="keyword">return</span>(features)</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">summarize_taxa &lt;- <span class="keyword">function</span>(features, taxonomy) &#123;</span><br><span class="line">	taxlevels&lt;-c(<span class="string">"Kingdom"</span>,<span class="string">"Phylum"</span>,<span class="string">"Class"</span>,<span class="string">"Order"</span>,</span><br><span class="line">                 <span class="string">"Family"</span>,<span class="string">"Genus"</span>,<span class="string">"Species"</span>)</span><br><span class="line">    <span class="keyword">if</span>(missing(features))&#123;</span><br><span class="line">        <span class="keyword">stop</span>(<span class="string">"Feature table not provided"</span>)</span><br><span class="line">    &#125;</span><br><span class="line">    <span class="keyword">if</span>(missing(taxonomy))&#123;</span><br><span class="line">        <span class="keyword">stop</span>(<span class="string">"taxonomy table not provided"</span>)</span><br><span class="line">    &#125;</span><br><span class="line">    <span class="keyword">if</span>(sum(colnames(taxonomy) %<span class="keyword">in</span>% taxlevels)!=<span class="number">7</span>)&#123;</span><br><span class="line">        <span class="keyword">stop</span>(<span class="string">"Taxonomy does not contain expected columns containing Kingdom,Phylum,Class,Order,Family,Genus,Species."</span>)</span><br><span class="line">    &#125;  </span><br><span class="line">    output&lt;-list()</span><br><span class="line">    <span class="keyword">for</span>(lvl <span class="keyword">in</span> taxlevels)&#123;</span><br><span class="line">    suppressMessages(</span><br><span class="line">      output[[lvl]]&lt;-</span><br><span class="line">        features %&gt;%</span><br><span class="line">        as.data.frame() %&gt;%</span><br><span class="line">        rownames_to_column(<span class="string">"FeatureID"</span>) %&gt;%</span><br><span class="line">        gather(-FeatureID, key=<span class="string">"SampleID"</span>, value=<span class="string">"Counts"</span>) %&gt;%</span><br><span class="line">        left_join(</span><br><span class="line">          taxonomy %&gt;% </span><br><span class="line">          rownames_to_column(<span class="string">"FeatureID"</span>) %&gt;%</span><br><span class="line">          unite(<span class="string">"Taxon"</span>,</span><br><span class="line">                taxlevels[<span class="number">1</span>:grep(lvl,taxlevels)],</span><br><span class="line">                sep=<span class="string">"; "</span>) %&gt;%</span><br><span class="line">          select(FeatureID, Taxon)</span><br><span class="line">        ) %&gt;%</span><br><span class="line">        group_by(SampleID, Taxon) %&gt;%</span><br><span class="line">        summarize(Counts=sum(Counts)) %&gt;%</span><br><span class="line">        ungroup() %&gt;%</span><br><span class="line">        spread(key=SampleID, value=Counts) %&gt;%</span><br><span class="line">        as.data.frame() %&gt;%</span><br><span class="line">        column_to_rownames(<span class="string">"Taxon"</span>)</span><br><span class="line">      )</span><br><span class="line">    &#125;</span><br><span class="line">    <span class="keyword">return</span>(output)</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line">taxa_heatmap&lt;-<span class="keyword">function</span>(features,metadata,</span><br><span class="line">                       category,normalize,ntoplot)&#123;</span><br><span class="line">  <span class="keyword">if</span>(missing(ntoplot) &amp; nrow(features)&gt;<span class="number">30</span>)&#123;</span><br><span class="line">      ntoplot=<span class="number">30</span></span><br><span class="line">  &#125; <span class="keyword">else</span> <span class="keyword">if</span> (missing(ntoplot))&#123;</span><br><span class="line">      ntoplot=nrow(features)</span><br><span class="line">  &#125;</span><br><span class="line">  <span class="keyword">if</span>(missing(normalize))&#123;normalize&lt;-<span class="string">"log10(percent)"</span>&#125;</span><br><span class="line">  <span class="keyword">if</span>(normalize==<span class="string">"log10(percent)"</span>)&#123;</span><br><span class="line">      features&lt;-log10(make_percent(features+<span class="number">1</span>))</span><br><span class="line">  &#125; <span class="keyword">else</span> <span class="keyword">if</span>(normalize==<span class="string">"clr"</span>)&#123;</span><br><span class="line">      features&lt;-make_clr(features)</span><br><span class="line">  &#125;</span><br><span class="line">  <span class="keyword">if</span>(missing(metadata))&#123;</span><br><span class="line">      metadata&lt;-data.frame(SampleID=colnames(features))</span><br><span class="line">  &#125;</span><br><span class="line">  <span class="keyword">if</span>(!<span class="string">"SampleID"</span> %<span class="keyword">in</span>% colnames(metadata))&#123;</span><br><span class="line">      metadata &lt;- metadata %&gt;% rownames_to_column(<span class="string">"SampleID"</span>)</span><br><span class="line">  &#125;</span><br><span class="line">  <span class="keyword">if</span>(!missing(category))&#123;</span><br><span class="line">    <span class="keyword">if</span>(!category %<span class="keyword">in</span>% colnames(metadata))&#123;</span><br><span class="line">        message(<span class="keyword">stop</span>(category,</span><br><span class="line">                     <span class="string">" not found as column in metdata"</span>))</span><br><span class="line">    &#125;</span><br><span class="line">  &#125;</span><br><span class="line">  plotfeats&lt;-names(sort(rowMeans(features),</span><br><span class="line">                        decreasing = <span class="literal">TRUE</span>)[<span class="number">1</span>:ntoplot]) </span><br><span class="line">  roworder&lt;-hclust(dist(features[plotfeats,]))</span><br><span class="line">  roworder&lt;-roworder$labels[roworder$order]</span><br><span class="line">  colorder&lt;-hclust(dist(t(features[plotfeats,])))</span><br><span class="line">  colorder&lt;-colorder$labels[colorder$order]</span><br><span class="line">  suppressMessages(</span><br><span class="line">  suppressWarnings(</span><br><span class="line">  fplot&lt;-</span><br><span class="line">    features %&gt;%</span><br><span class="line">    as.data.frame() %&gt;%</span><br><span class="line">    rownames_to_column(<span class="string">"Taxon"</span>) %&gt;%</span><br><span class="line">    gather(-Taxon, key=<span class="string">"SampleID"</span>, value=<span class="string">"Abundance"</span>) %&gt;%</span><br><span class="line">    filter(Taxon %<span class="keyword">in</span>% plotfeats) %&gt;%</span><br><span class="line">    mutate(Taxon=factor(Taxon, levels=rev(plotfeats))) %&gt;%</span><br><span class="line">    left_join(metadata) %&gt;%</span><br><span class="line">    mutate(Taxon=factor(Taxon, levels=roworder)) %&gt;%</span><br><span class="line">    mutate(SampleID=factor(SampleID, levels=colorder))</span><br><span class="line">  ))</span><br><span class="line"></span><br><span class="line">  bplot&lt;-</span><br><span class="line">    ggplot(fplot, aes(x=SampleID, y=Taxon, fill=Abundance)) +</span><br><span class="line">    geom_tile(stat=<span class="string">"identity"</span>) +</span><br><span class="line">    theme_q2r() +</span><br><span class="line">    theme(axis.text.x = element_text(angle=<span class="number">45</span>, hjust=<span class="number">1</span>)) +</span><br><span class="line">    coord_cartesian(expand=<span class="literal">FALSE</span>) +</span><br><span class="line">    xlab(<span class="string">"Sample"</span>) +</span><br><span class="line">    ylab(<span class="string">"Feature"</span>) +</span><br><span class="line">    scale_fill_viridis_c()</span><br><span class="line">  <span class="keyword">if</span>(!missing(category))&#123;</span><br><span class="line">      bplot&lt;-bplot + </span><br><span class="line">      facet_grid(~get(category),scales=<span class="string">"free_x"</span>,space=<span class="string">"free"</span>)</span><br><span class="line">  &#125;</span><br><span class="line">  <span class="keyword">return</span>(bplot)</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line">SVs&lt;-read_qza(<span class="string">"3_ASV/table.qza"</span>)$data</span><br><span class="line">taxonomy&lt;-read_qza(<span class="string">"4_taxonomy/taxonomy.qza"</span>)$data %&gt;% </span><br><span class="line">parse_taxonomy()</span><br><span class="line"></span><br><span class="line">taxasums&lt;-summarize_taxa(SVs, taxonomy)$Genus</span><br><span class="line">taxa_heatmap(taxasums, metadata, <span class="string">"Treat1"</span>)</span><br><span class="line"></span><br><span class="line">ggsave(<span class="string">"heatmap.pdf"</span>, height=<span class="number">4</span>, width=<span class="number">8</span>, device=<span class="string">"pdf"</span>)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment">## 物种柱状图</span></span><br><span class="line"></span><br><span class="line">taxa_barplot&lt;-<span class="keyword">function</span>(features,metadata,category,normalize,ntoplot)&#123;</span><br><span class="line">  q2r_palette&lt;-c(</span><br><span class="line">    <span class="string">"blue4"</span>,</span><br><span class="line">    <span class="string">"olivedrab"</span>,</span><br><span class="line">    <span class="string">"firebrick"</span>,</span><br><span class="line">    <span class="string">"gold"</span>,</span><br><span class="line">    <span class="string">"darkorchid"</span>,</span><br><span class="line">    <span class="string">"steelblue2"</span>,</span><br><span class="line">    <span class="string">"chartreuse1"</span>,</span><br><span class="line">    <span class="string">"aquamarine"</span>,</span><br><span class="line">    <span class="string">"yellow3"</span>,</span><br><span class="line">    <span class="string">"coral"</span>,</span><br><span class="line">    <span class="string">"grey"</span></span><br><span class="line">  )</span><br><span class="line">  </span><br><span class="line">  <span class="keyword">if</span>(missing(ntoplot) &amp; nrow(features)&gt;<span class="number">10</span>)&#123;</span><br><span class="line">      ntoplot=<span class="number">10</span></span><br><span class="line">  &#125; <span class="keyword">else</span> <span class="keyword">if</span> (missing(ntoplot))&#123;</span><br><span class="line">      ntoplot=nrow(features)</span><br><span class="line">  &#125;</span><br><span class="line">  <span class="keyword">if</span>(missing(normalize))&#123;</span><br><span class="line">      normalize&lt;-<span class="string">"percent"</span></span><br><span class="line">  &#125;</span><br><span class="line">  <span class="keyword">if</span>(normalize==<span class="string">"percent"</span>)&#123;</span><br><span class="line">      features&lt;-make_percent(features)</span><br><span class="line">  &#125; <span class="keyword">else</span> <span class="keyword">if</span>(normalize==<span class="string">"proportion"</span>)&#123;</span><br><span class="line">      features&lt;-make_proportion(features)</span><br><span class="line">  &#125;</span><br><span class="line">  <span class="keyword">if</span>(missing(metadata))&#123;</span><br><span class="line">      metadata&lt;-data.frame(SampleID=colnames(features))</span><br><span class="line">  &#125;</span><br><span class="line">  <span class="keyword">if</span>(!<span class="string">"SampleID"</span> %<span class="keyword">in</span>% colnames(metadata))&#123;</span><br><span class="line">      metadata &lt;- metadata %&gt;% rownames_to_column(<span class="string">"SampleID"</span>)</span><br><span class="line">  &#125;</span><br><span class="line">  <span class="keyword">if</span>(!missing(category))&#123;</span><br><span class="line">    <span class="keyword">if</span>(!category %<span class="keyword">in</span>% colnames(metadata))&#123;</span><br><span class="line">        message(<span class="keyword">stop</span>(category, <span class="string">" not found as column in metdata"</span>))</span><br><span class="line">    &#125;</span><br><span class="line">  &#125;</span><br><span class="line"></span><br><span class="line">  plotfeats&lt;-names(sort(rowMeans(features),</span><br><span class="line">                        decreasing=<span class="literal">TRUE</span>)[<span class="number">1</span>:ntoplot]) </span><br><span class="line">  suppressMessages(</span><br><span class="line">  suppressWarnings(</span><br><span class="line">  fplot&lt;-</span><br><span class="line">    features %&gt;%</span><br><span class="line">    as.data.frame() %&gt;%</span><br><span class="line">    rownames_to_column(<span class="string">"Taxon"</span>) %&gt;%</span><br><span class="line">    gather(-Taxon, key=<span class="string">"SampleID"</span>, </span><br><span class="line">           value=<span class="string">"Abundance"</span>) %&gt;%</span><br><span class="line">    mutate(Taxon=if_else(Taxon %<span class="keyword">in</span>% plotfeats,</span><br><span class="line">                         Taxon,<span class="string">"Remainder"</span>)) %&gt;%</span><br><span class="line">    group_by(Taxon, SampleID) %&gt;%</span><br><span class="line">    summarize(Abundance=sum(Abundance)) %&gt;%</span><br><span class="line">    ungroup() %&gt;%</span><br><span class="line">    mutate(Taxon=factor(Taxon, </span><br><span class="line">                        levels=rev(c(plotfeats,<span class="string">"Remainder"</span>)))) %&gt;%</span><br><span class="line">    left_join(metadata)</span><br><span class="line">  ))</span><br><span class="line"></span><br><span class="line">  bplot&lt;-</span><br><span class="line">    ggplot(fplot, aes(x=SampleID, y=Abundance, fill=Taxon)) +</span><br><span class="line">    geom_bar(stat=<span class="string">"identity"</span>) +</span><br><span class="line">    theme_q2r() +</span><br><span class="line">    theme(axis.text.x = element_text(angle=<span class="number">45</span>, hjust=<span class="number">1</span>)) +</span><br><span class="line">    coord_cartesian(expand=<span class="literal">FALSE</span>) +</span><br><span class="line">    xlab(<span class="string">"Sample"</span>) +</span><br><span class="line">    ylab(<span class="string">"Abundance"</span>)</span><br><span class="line">  <span class="keyword">if</span>(ntoplot&lt;=<span class="number">10</span>)&#123;</span><br><span class="line">      bplot&lt;-bplot+scale_fill_manual(values=rev(q2r_palette),</span><br><span class="line">                                     name=<span class="string">""</span>)</span><br><span class="line">  &#125;</span><br><span class="line">  <span class="keyword">if</span>(!missing(category))&#123;</span><br><span class="line">      bplot&lt;-bplot+facet_grid(~get(category),</span><br><span class="line">                              scales=<span class="string">"free_x"</span>,space=<span class="string">"free"</span>)</span><br><span class="line">  &#125;</span><br><span class="line">  <span class="keyword">return</span>(bplot)</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line">taxa_barplot(taxasums, metadata, <span class="string">"Treat1"</span>)</span><br><span class="line"></span><br><span class="line">ggsave(<span class="string">"barplot.pdf"</span>, height=<span class="number">4</span>, width=<span class="number">8</span>, device=<span class="string">"pdf"</span>)</span><br></pre></td></tr></table></figure>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 退出 R</span></span><br><span class="line"></span><br><span class="line">q()</span><br></pre></td></tr></table></figure>
<h2><span id="附构建注释数据库">附：构建注释数据库</span></h2>
<p>第一次使用，需要先构建 classify-sklearn 专用的注释数据库，也可以去官网下载。</p>
<ul>
<li><a href="https://data.qiime2.org/2020.8/common/silva-138-99-nb-classifier.qza" target="_blank" rel="noopener">Silva 138 99% OTUs full-length sequences</a> (MD5: <code>fddefff8bfa2bbfa08b9cad36bcdf709</code>)</li>
<li><a href="https://data.qiime2.org/2020.8/common/silva-138-99-515-806-nb-classifier.qza" target="_blank" rel="noopener">Silva 138 99% OTUs from 515F/806R region of sequences</a> (MD5: <code>28105eb0f1256bf38b9bb310c701dc4e</code>)</li>
<li><a href="https://data.qiime2.org/2020.8/common/gg-13-8-99-nb-classifier.qza" target="_blank" rel="noopener">Greengenes 13_8 99% OTUs full-length sequences</a> (MD5: <code>03078d15b265f3d2d73ce97661e370b1</code>)</li>
<li><a href="https://data.qiime2.org/2020.8/common/gg-13-8-99-515-806-nb-classifier.qza" target="_blank" rel="noopener">Greengenes 13_8 99% OTUs from 515F/806R region of sequences</a> (MD5: <code>682be39339ef36a622b363b8ee2ff88b</code>)</li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">## 耗内存，需要提高docker的内存配置,silva 预计可能需要超过 8G</span></span><br><span class="line"></span><br><span class="line">mkdir -p training-feature-classifiers</span><br><span class="line"><span class="built_in">cd</span> training-feature-classifiers</span><br><span class="line"></span><br><span class="line">cp /data/qiime2/data/02_data/99_otus.fasta .   </span><br><span class="line">cp /data/qiime2/data/02_data/99_otu_taxonomy.txt .     </span><br><span class="line"></span><br><span class="line">qiime tools import \</span><br><span class="line">  --<span class="built_in">type</span> <span class="string">'FeatureData[Sequence]'</span> \</span><br><span class="line">  --input-path 99_otus.fasta  \</span><br><span class="line">  --output-path 99_otus.qza </span><br><span class="line"></span><br><span class="line">qiime tools import \</span><br><span class="line">  --<span class="built_in">type</span> <span class="string">'FeatureData[Taxonomy]'</span> \</span><br><span class="line">  --input-format HeaderlessTSVTaxonomyFormat \</span><br><span class="line">  --input-path 99_otu_taxonomy.txt \</span><br><span class="line">  --output-path ref-taxonomy.qza</span><br><span class="line"></span><br><span class="line"><span class="comment">#16SV4V5</span></span><br><span class="line">qiime feature-classifier extract-reads \</span><br><span class="line">  --i-sequences 99_otus.qza  \</span><br><span class="line">  --p<span class="_">-f</span>-primer <span class="variable">$front_f</span> \</span><br><span class="line">  --p-r-primer  <span class="variable">$front_r</span>  \</span><br><span class="line">  --p-trunc-len 0 \</span><br><span class="line">  --p-min-length 100 \</span><br><span class="line">  --p-max-length 500 \</span><br><span class="line">  --o-reads ref-seqs.qza</span><br><span class="line">  </span><br><span class="line">qiime feature-classifier fit-classifier-naive-bayes \</span><br><span class="line">  --i-reference-reads ref-seqs.qza \</span><br><span class="line">  --i-reference-taxonomy ref-taxonomy.qza \</span><br><span class="line">  --o-classifier gg_13_8_99_16S_V4V5_classifier.qza</span><br></pre></td></tr></table></figure>
<p><img src="https://gitee.com/microbiomecat/image_hosting/raw/master/img/image-20200923210736980.png" alt="image-20200923210736980"></p>
<script type="text/javascript" src="https://cdn.jsdelivr.net/npm/kity@2.0.4/dist/kity.min.js"></script><script type="text/javascript" src="https://cdn.jsdelivr.net/npm/kityminder-core@1.4.50/dist/kityminder.core.min.js"></script><script defer="true" type="text/javascript" src="https://cdn.jsdelivr.net/npm/hexo-simple-mindmap@0.2.0/dist/mindmap.min.js"></script><link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/npm/hexo-simple-mindmap@0.2.0/dist/mindmap.min.css">
    </div>

      <div>
        
          
<div>
    
        <div style="text-align:center;color: #ccc;font-size:20px;">-----------------------------本文结束  <i class="fa fa-paw"></i>  感谢您的阅读-----------------------------</div>
    
</div>
        
      </div>
      
    
    
    
      
  <div class="popular-posts-header">相关文章</div>
  <ul class="popular-posts">
    <li class="popular-posts-item">
      <div class="popular-posts-title"><a href="\20200726\QIIME2-based-amplicon-analysis-tutorial(Preparation)\" rel="bookmark">基于 QIIME2 的扩增子分析教程-准备篇</a></div>
    </li>
    <li class="popular-posts-item">
      <div class="popular-posts-title"><a href="\20201005\QIIME2-based-amplicon-analysis-tutorial(Basic)\" rel="bookmark">基于 QIIME2 的扩增子分析教程-基础篇</a></div>
    </li>
  </ul>

        <div class="reward-container">
  <div><p style="font-size:14px; color:#34495e; margin:0 0 5px 0;">赞赏一下吧～.</div>
  <button onclick="var qr = document.getElementById('qr'); qr.style.display = (qr.style.display === 'none') ? 'block' : 'none';">
    打赏
  </button>
  <div id="qr" style="display: none;">
      
      <div style="display: inline-block;">
        <img src="/./../image/reward1.jpg" alt="Sansan Li 微信支付">
        <p>微信支付</p>
      </div>
      
      <div style="display: inline-block;">
        <img src="/./../image/reward2.png" alt="Sansan Li 支付宝">
        <p>支付宝</p>
      </div>

  </div>
</div>

        

<div>
<ul class="post-copyright">
  <li class="post-copyright-author">
    <strong>本文作者： </strong>Sansan Li
  </li>
  <li class="post-copyright-link">
    <strong>本文链接：</strong>
    <a href="https://www.microbiomecat.xyz/20201006/QIIME2-based-amplicon-analysis-tutorial(Practical)/" title="基于 QIIME2 的扩增子分析教程-实战篇">https://www.microbiomecat.xyz/20201006/QIIME2-based-amplicon-analysis-tutorial(Practical)/</a>
  </li>
  <li class="post-copyright-license">
    <strong>版权声明： </strong>本博客所有文章除特别声明外，均采用 <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" rel="noopener" target="_blank"><i class="fab fa-fw fa-creative-commons"></i>BY-NC-SA</a> 许可协议。转载请注明出处！
  </li>
</ul>
</div>





      <footer class="post-footer">
          
          <div class="post-tags">
              <a href="/tags/QIIME2/" rel="tag"><i class="fa fa-tag"></i> QIIME2</a>
              <a href="/tags/%E6%89%A9%E5%A2%9E%E5%AD%90/" rel="tag"><i class="fa fa-tag"></i> 扩增子</a>
          </div>

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/20201005/QIIME2-based-amplicon-analysis-tutorial(Basic)/" rel="prev" title="基于 QIIME2 的扩增子分析教程-基础篇">
      <i class="fa fa-chevron-left"></i> 基于 QIIME2 的扩增子分析教程-基础篇
    </a></div>
      <div class="post-nav-item"></div>
    </div>
      </footer>
    
  </article>
  
  
  



          </div>
          
    <div class="comments" id="valine-comments"></div>

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          文章目录
        </li>
        <li class="sidebar-nav-overview">
          站点概览
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-2"><a class="nav-link" href="#"><span class="nav-number">1.</span> <span class="nav-text">摘要</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#"><span class="nav-number">2.</span> <span class="nav-text">一、获取数据</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#"><span class="nav-number">3.</span> <span class="nav-text">二、参数汇总</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#"><span class="nav-number">4.</span> <span class="nav-text">三、创建manifest文件</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#"><span class="nav-number">5.</span> <span class="nav-text">(四)、一个文库的操作演示</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#"><span class="nav-number">5.1.</span> <span class="nav-text">数据导入 QIIME2</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#"><span class="nav-number">5.2.</span> <span class="nav-text">切除引物</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#"><span class="nav-number">5.3.</span> <span class="nav-text">DADA2</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#"><span class="nav-number">6.</span> <span class="nav-text">四、一个或多个文库</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#"><span class="nav-number">7.</span> <span class="nav-text">五、合并特征序列与特征表格</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#"><span class="nav-number">8.</span> <span class="nav-text">六、建树</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#"><span class="nav-number">9.</span> <span class="nav-text">七、物种注释</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#"><span class="nav-number">10.</span> <span class="nav-text">八、核心多样性指数</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#"><span class="nav-number">11.</span> <span class="nav-text">十、qiime2R 可视化探索</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#"><span class="nav-number">11.1.</span> <span class="nav-text">数据导入</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#"><span class="nav-number">11.2.</span> <span class="nav-text">绘图</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#"><span class="nav-number">12.</span> <span class="nav-text">附：构建注释数据库</span></a></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
    <img class="site-author-image" itemprop="image" alt="Sansan Li"
      src="/image/876364972032900262.jpg">
  <p class="site-author-name" itemprop="name">Sansan Li</p>
  <div class="site-description" itemprop="description">MicrobiomeCAT -- 微生物组学猫</div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives">
        
          <span class="site-state-item-count">8</span>
          <span class="site-state-item-name">日志</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
            <a href="/categories">
          
        <span class="site-state-item-count">4</span>
        <span class="site-state-item-name">分类</span></a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/tags">
          
        <span class="site-state-item-count">8</span>
        <span class="site-state-item-name">标签</span></a>
      </div>
  </nav>
</div>
  <div class="cc-license motion-element" itemprop="license">
    <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" class="cc-opacity" rel="noopener" target="_blank"><img src="/images/cc-by-nc-sa.svg" alt="Creative Commons"></a>
  </div>



      </div>
        <div class="back-to-top motion-element">
          <i class="fa fa-arrow-up"></i>
          <span>0%</span>
        </div>
      
          <script type="text/javascript" charset="utf-8" src="/js/tagcloud.js"></script>
          <script type="text/javascript" charset="utf-8" src="/js/tagcanvas.js"></script>
          <div class="widget-wrap">
              <div id="myCanvasContainer" class="widget tagcloud">
              <canvas width="220" height="250" id="resCanvas" style="width=100%">
                  <ul class="tag-list" itemprop="keywords"><li class="tag-list-item"><a class="tag-list-link" href="/tags/Docker/" rel="tag">Docker</a><span class="tag-list-count">1</span></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/QIIME2/" rel="tag">QIIME2</a><span class="tag-list-count">3</span></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/hexo/" rel="tag">hexo</a><span class="tag-list-count">2</span></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/markdown/" rel="tag">markdown</a><span class="tag-list-count">1</span></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/%E6%89%A9%E5%A2%9E%E5%AD%90/" rel="tag">扩增子</a><span class="tag-list-count">3</span></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/%E6%94%B6%E7%BA%B3/" rel="tag">收纳</a><span class="tag-list-count">1</span></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/%E8%A3%85%E4%BF%AE/" rel="tag">装修</a><span class="tag-list-count">2</span></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/%E8%AF%BB%E4%B9%A6%E7%AC%94%E8%AE%B0/" rel="tag">读书笔记</a><span class="tag-list-count">2</span></li></ul>
              </canvas>
              </div>
          </div>
        
    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 
  <span itemprop="copyrightYear">2020</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Sansan Li</span>
    <span class="post-meta-divider">|</span>
    <span class="post-meta-item-icon">
      <i class="fa fa-chart-area"></i>
    </span>
    <span title="站点总字数">56k</span>
    <span class="post-meta-divider">|</span>
    <span class="post-meta-item-icon">
      <i class="fa fa-coffee"></i>
    </span>
    <span title="站点阅读时长">51 分钟</span>
</div>

        
<div class="busuanzi-count">
  <script async src="https://busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script>
    <span class="post-meta-item" id="busuanzi_container_site_uv" style="display: none;">
      <span class="post-meta-item-icon">
        <i class="fa fa-user"></i>
      </span>
      <span class="site-uv" title="总访客量">
        <span id="busuanzi_value_site_uv"></span>
      </span>
    </span>
    <span class="post-meta-divider">|</span>
    <span class="post-meta-item" id="busuanzi_container_site_pv" style="display: none;">
      <span class="post-meta-item-icon">
        <i class="fa fa-eye"></i>
      </span>
      <span class="site-pv" title="总访问量">
        <span id="busuanzi_value_site_pv"></span>
      </span>
    </span>
</div>








      </div>
    </footer>
  </div>

  
  <script src="/lib/anime.min.js"></script>
  <script src="//cdn.jsdelivr.net/npm/jquery@3/dist/jquery.min.js"></script>
  <script src="//cdn.jsdelivr.net/gh/fancyapps/fancybox@3/dist/jquery.fancybox.min.js"></script>
  <script src="//cdn.jsdelivr.net/npm/medium-zoom@1/dist/medium-zoom.min.js"></script>
  <script src="/lib/velocity/velocity.min.js"></script>
  <script src="/lib/velocity/velocity.ui.min.js"></script>

<script src="/js/utils.js"></script>

<script src="/js/motion.js"></script>


<script src="/js/schemes/pisces.js"></script>


<script src="/js/next-boot.js"></script>

<script src="/js/bookmark.js"></script>




  
  <script>
    (function(){
      var canonicalURL, curProtocol;
      //Get the <link> tag
      var x=document.getElementsByTagName("link");
		//Find the last canonical URL
		if(x.length > 0){
			for (i=0;i<x.length;i++){
				if(x[i].rel.toLowerCase() == 'canonical' && x[i].href){
					canonicalURL=x[i].href;
				}
			}
		}
    //Get protocol
	    if (!canonicalURL){
	    	curProtocol = window.location.protocol.split(':')[0];
	    }
	    else{
	    	curProtocol = canonicalURL.split(':')[0];
	    }
      //Get current URL if the canonical URL does not exist
	    if (!canonicalURL) canonicalURL = window.location.href;
	    //Assign script content. Replace current URL with the canonical URL
      !function(){var e=/([http|https]:\/\/[a-zA-Z0-9\_\.]+\.baidu\.com)/gi,r=canonicalURL,t=document.referrer;if(!e.test(r)){var n=(String(curProtocol).toLowerCase() === 'https')?"https://sp0.baidu.com/9_Q4simg2RQJ8t7jm9iCKT-xh_/s.gif":"//api.share.baidu.com/s.gif";t?(n+="?r="+encodeURIComponent(document.referrer),r&&(n+="&l="+r)):r&&(n+="?l="+r);var i=new Image;i.src=n}}(window);})();
  </script>




  
<script src="/js/local-search.js"></script>













  

  


<script>
NexT.utils.loadComments(document.querySelector('#valine-comments'), () => {
  NexT.utils.getScript('https://cdn.jsdelivr.net/npm/valine@1.4.14/dist/Valine.min.js', () => {
    var GUEST = ['nick', 'mail', 'link'];
    var guest = 'nick,mail,link';
    guest = guest.split(',').filter(item => {
      return GUEST.includes(item);
    });
    new Valine({
      el         : '#valine-comments',
      verify     : false,
      notify     : true,
      appId      : 'GItGRzUdbysXj6n5xqi5dRgu-MdYXbMMI',
      appKey     : 'r68FaoVS1t6dfi2SdVhYhfq6',
      placeholder: "向我提问吧",
      avatar     : 'mm',
      meta       : guest,
      pageSize   : '10' || 10,
      visitor    : false,
      lang       : 'zh-cn' || 'zh-cn',
      path       : location.pathname,
      recordIP   : false,
      serverURLs : ''
    });
  }, window.Valine);
});
</script>


  <!-- Me added -->
<script src="/js/counttime.js"></script>
</body>
</html>
